src/connection-manager/index.ts

```diff
@@ -208,9 +220,43 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
   async afterStart () {
     this.components.getUpgrader().addEventListener('connection', this.onConnect)
     this.components.getUpgrader().addEventListener('connectionEnd', this.onDisconnect)
+
+    // re-connect to any peers with the `keep-alive` tag
+    void Promise.resolve()
+      .then(async () => {
+        const keepAlivePeers: PeerId[] = []
+
+        for (const peer of await this.components.getPeerStore().all()) {
+          const tags = await this.components.getPeerStore().getTags(peer.id)
+          const hasKeepAlive = tags.filter(tag => tag.name === 'keep-alive').length > 0
```

Comment:
Can you define and export the string as a constant? I think downstream users should tag via exported constants if possible to avoid typos and confusion.

Reply:
I've opened this one: https://github.com/libp2p/js-libp2p-interfaces/pull/263

Reply:
Mind you, we want to be careful about different modules reusing tag names - otherwise one module could decide a given peer is not important any more and remove the tag, a different module would then be rug-pulled and the bug would be quite hard to track down.